syz-manager/manager.go

```diff
@@ -584,10 +586,17 @@ func (mgr *Manager) runInstance(index int) (*Crash, error) {
 		log.Logf(0, "vm-%v: running for %v, restarting", index, time.Since(start))
 		return nil, nil
 	}
+
+	instanceName := fmt.Sprintf("vm-%d", index)
+	var machineInfo []byte
+	if info, ok := mgr.serv.getMachineInfo(instanceName); ok {
```

Comment:
This needs to happen after we do `inst.Close`, otherwise there is some potential for races + we can improve getMachineInfo.

Reply:
I used a anonymous function to provide a scope for defer to run so that we can share variables (`index`, `instanceName`) easily. Not sure if this is idiomatic in go. If not I will refactor it to a new function.